core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -88,11 +88,22 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
 
   def map[D](f: B => D)(implicit F: Functor[F]): EitherT[F, A, D] = bimap(identity, f)
 
+  def mapF[N[_], D](fe: F[Either[A, B]] => N[Either[A, D]]): EitherT[N, A, D] = EitherT(fe(value))
```

Comment:
This transform sounds like we might need a new higher order type class? like 
```scala
trait HFunctor[F[_[_], _] {
  def hmap[A[_], B[_], T](fa: F[A, T])(fk: A ~> B): F[B, T]
}
```
There is some discussion in this [gist](https://gist.github.com/kailuowang/ddf4f99a3bd97f150277a0c614e0a6f2). It maybe too theoretical, but I think we can see its practial usage here.